plugins/elasticsearch/middleware.go

```diff
@@ -229,13 +229,13 @@ func intercept(h http.HandlerFunc) http.HandlerFunc {
 		for _, index := range indices {
 			alias := classify.GetIndexAlias(index)
 			if alias != "" {
-				body = bytes.Replace(body, []byte(index), []byte(alias), -1)
+				body = bytes.Replace(body, []byte(`"`+index+`"`), []byte(`"`+alias+`"`), -1)
 				continue
 			}
 			// if alias is present in url get index name from cache
 			indexName := classify.GetAliasIndex(index)
 			if indexName != "" {
-				body = bytes.Replace(body, []byte(indexName), []byte(index), -1)
+				body = bytes.Replace(body, []byte(`"`+indexName+`"`), []byte(`"`+index+`"`), -1)
```

Comment:
@lakhansamani I'm not getting this part, can you please explain?

Reply:
To reduce the accidental replacement of index name in any other string, we decide to replace the index name with quotes. This atleast narrows down the side effect if any, example 
`"products_reindexed_1"` = `"products"`

Reply:
@bietkul I get that but what are you doing in this line?
I can understand the  `index` to `alias` replacement but what are you doing here?

Reply:
@bietkul It looks good to me.
1. We do byte replacement instead of marshalling / unmarshalling the response and replacing the specific field for performance reason.
2. Since we byte replacement, it's going to find every occurance of the search term and replace it with the replace term. To avoid side-effects, @lakhansamani is now using a narrower match condition so that only when the search term is found in the pattern `"${search}"`, it gets replaced. A mere presence of ${search} without surrounding quotes shouldn't get replaced (a user can be storing an index name as part of one of the fields).